backtracking/generate_parentheses.cpp

```diff
@@ -0,0 +1,112 @@
+/**
+ * @file
+ * @brief Well-formed [Generated
+ Parentheses](https://leetcode.com/explore/interview/card/top-interview-questions-medium/109/backtracking/794/) with all combinations.
+ *
+ * @details a sequence of parentheses is well-formed if each opening parentheses
+ * has a corresponding closing parenthesis
+ * and the closing parentheses are correctly ordered
+ *
+ * @author [Giuseppe Coco](https://github.com/WoWS17)
+
+ */
+
+#include <cassert>   /// for assert
+#include <iostream>  /// for I/O operation
+#include <vector>    /// for vector container
+
+/**
+ * @brief Backtracking algorithms
+ * @namespace backtracking
+ */
+namespace backtracking {
+/**
+ * @brief generate_parentheses class
+ */
+class generate_parentheses {
+ private:
+    std::vector<std::string> res;  ///< Contains all possible valid patterns
+
+    void makeStrings(std::string str, int n, int closed, int open);
+
+ public:
+    std::vector<std::string> generate(int n);
+};
+}  // namespace backtracking
+
+/**
+ * @brief function that implements backtracking
+ *
+ * @param str string build during backtracking
+ * @param n number of pairs of parentheses
+ * @param closed number of closed parentheses
+ * @param open number of open parentheses
+ */
+
+void backtracking::generate_parentheses::makeStrings(std::string str, int n,
+                                                     int closed, int open) {
+    if (closed > open)  // We can never have more closed than open
+        return;
+
+    if (str.length() == 2 * n and
+        closed != open)  // closed and open must be the same
```

Comment:
`and` should not be used in C++, I don't even think it's possible or works.
```suggestion
    if ((str.length() == 2 * n) &&
        (closed != open)) {  // closed and open must be the same
```

Reply:
it actually works

Reply:
Do you know what Awesome CI Workflow / Compile checks (windows-latest) doesn't work?